utilities/gardenlinux-bumper/bumper.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import re
+import requests
+
+
+def get_latest_release() -> (int, int, str):
+    gcp_vm_re = re.compile(r'^gcp-gardener_prod-amd64-([0-9]+)\.([0-9]+)-([0-9a-f]+)\.tar\.xz$')
+    url = "https://api.github.com/repos/gardenlinux/gardenlinux/releases/latest"
+    headers = {
+        'Accept': 'application/vnd.github+json',
+        'Authorization': f'Bearer {os.environ["GITHUB_TOKEN"]}',
+        'X-GitHub-Api-Version': '2022-11-28',
+    }
+    response = requests.get(url, headers=headers)
+    response.raise_for_status()
+    latest_release = response.json()
+
+    for a in latest_release['assets']:
+        match = gcp_vm_re.match(a['name'])
+        if match:
+            major = int(match[1])
+            minor = int(match[2])
+            checksum = match[3]
+
+            return (major, minor, checksum)
+    raise RuntimeError("Failed to find the GCP VM asset")
+
+
+def get_current_version(image_file: str) -> (int, int, str):
+    gcp_vm_re = re.compile(r'^gardenlinux-gcp-gardener-prod-amd64-([0-9]+)-([0-9]+)-([0-9a-f]+)$')
+
+    with open(image_file, 'r') as f:
+        image = f.readline()
+        match = gcp_vm_re.match(image)
+        if match is None:
+            raise RuntimeError('Configured image did not match')
+
+        major = int(match[1])
+        minor = int(match[2])
+        checksum = match[3]
+
+        return (major, minor, checksum)
+
+
+def get_gardenlinux_image(version_data: (int, int, str)) -> str:
+    major = version_data[0]
+    minor = version_data[1]
+    checksum = version_data[2]
+
+    return f'gardenlinux-gcp-gardener-prod-amd64-{major}-{minor}-{checksum}'
+
+
+def image_is_outdated(latest: (int, int, str), current: (int, int, str)):
+    latest_major = latest[0]
+    latest_minor = latest[1]
+
+    current_major = current[0]
+    current_minor = current[1]
+
+    return latest_major > current_major or (latest_major == current_major and latest_minor > current_minor)
```

Comment:
[nit] this can be slightly simplified by using tuple comparison

```
return (latest[0], latest[1]) > (current[0], current[1])
```

Reply:
I can probably use a sub-tuple instead, not sure what you think about readability though
```python
return latest[:2] > current[:2]
```